### PR TITLE
SlowLog: Update `Wait_TS` field in slow log for every sql 

### LINKED ISSUE #39713: The field Wait_ts is not updated in slow log for the sqls which are in the same transaction
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

set tidb_slow_log_threshold = 0
begin;
select * from t1 where id1 = 1;
select * from t1 where id2>3;

### 2. What did you expect to see? (Required)
The two sqls above have different Wait_TS in slow log.

### 3. What did you see instead (Required)
The two sqls above have the same Wait_TS in slow log.

### 4. What is your TiDB version? (Required)

Release Version: v6.5.0-alpha-272-g35819ee
Edition: Community
Git Commit Hash: 35819ee43ffd932d8b1dd3c6f8804239dd5ea3aa
Git Branch: master
UTC Build Time: 2022-12-07 07:52:28
GoVersion: go1.19
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: tikv

### PR DESCRIPTION
**Pull request by TonsnakeLin**
### What problem does this PR solve?

Issue Number: close #39713

Problem Summary:
The field `Wait_TS` in slow log is always the wait time of `begin` statement for all sqls in a transaction. It will mislead the user to diagnose slow queries.

### What is changed and how it works?
This PR will update the wait time of `tsoRequeset` for every sql in a transaction.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: sessiontxn/txn_rc_tso_optimize_test.go [hunk 1]
```diff
@@ -790,3 +790,34 @@ func TestConflictErrorsUseRcWriteCheckTs(t *testing.T) {
 
 	require.NoError(t, failpoint.Disable("github.com/pingcap/tidb/executor/assertPessimisticLockErr"))
 }
+
+func TestRcWaitTSInSlowLog(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+
+	// ctx := context.Background()
```
Comment (sticnarf) [inline sessiontxn/txn_rc_tso_optimize_test.go:797]: Remove

**Review by sticnarf — APPROVED** [review]

File: executor/executor.go [hunk 2]
```diff
@@ -2167,6 +2167,7 @@ func ResetContextOfStmt(ctx sessionctx.Context, s ast.StmtNode) (err error) {
 	vars.ClearStmtVars()
 	vars.PrevFoundInBinding = vars.FoundInBinding
 	vars.FoundInBinding = false
+	vars.DurationWaitTS = 0
```
Comment (cfzjywxk) [inline executor/executor.go:2170]: Where is it used for slow log generation or information filling?

**Review by cfzjywxk — APPROVED** [review]

**Comment by TonsnakeLin** [comment]
@cfzjywxk please merge again, Thank you

**Pull request merged**